src/Domain/LogicalCoordinates.cpp

```diff
@@ -2,57 +2,66 @@
 // See LICENSE.txt for details.
 
 /// \file
-/// Defines functions logical_coordinates and interface_grid_coordinates
+/// Defines functions logical_coordinates and interface_logical_coordinates
 
 #include "Domain/LogicalCoordinates.hpp"
 
 #include <array>
 
 #include "DataStructures/DataVector.hpp"
-#include "DataStructures/Index.hpp"  // IWYU pragma: keep
 #include "DataStructures/IndexIterator.hpp"
+#include "DataStructures/Mesh.hpp"           // IWYU pragma: keep
 #include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
 #include "Domain/Direction.hpp"              // IWYU pragma: keep
 #include "Domain/Side.hpp"
-#include "NumericalAlgorithms/Spectral/LegendreGaussLobatto.hpp"
+#include "NumericalAlgorithms/Spectral/Spectral.hpp"
 #include "Utilities/Gsl.hpp"
 
 template <size_t VolumeDim>
 tnsr::I<DataVector, VolumeDim, Frame::Logical> logical_coordinates(
-    const Index<VolumeDim>& extents) noexcept {
-  tnsr::I<DataVector, VolumeDim, Frame::Logical> logical_x(extents.product());
+    const Mesh<VolumeDim>& mesh) noexcept {
+  tnsr::I<DataVector, VolumeDim, Frame::Logical> logical_x(
+      mesh.number_of_grid_points());
   for (size_t d = 0; d < VolumeDim; ++d) {
     const auto& collocation_points_in_this_dim =
-        Basis::lgl::collocation_points(extents[d]);
-    for (IndexIterator<VolumeDim> index(extents); index; ++index) {
+        Spectral::collocation_points(mesh.slice_through(d));
+    for (IndexIterator<VolumeDim> index(mesh.extents()); index; ++index) {
       logical_x.get(d)[index.collapsed_index()] =
           collocation_points_in_this_dim[index()[d]];
     }
   }
   return logical_x;
 }
 
+template <>
```

Comment:
why is this here?

Reply:
Are you asking why I added the 1D specialization? `Mesh<Dim>::slice_through()` is available only for `Dim>0`.

And yes, it compiles fine.

Reply:
Ah, it's in the cpp file. I was wondering how it new about the function. I'd find it clearer to have it below the general implementation personally. It might be useful to add a short comment saying the mesh slice through thing.

Reply:
👍 I'll do that